ionics_fits/common.py

```diff
@@ -737,17 +732,19 @@ def _fit_significance(self) -> Optional[float]:
         return None
 
     def evaluate(
-        self, x_fit: Optional[Union[Array[("num_samples",), np.float64], int]] = None
+        self,
+        x_fit: Optional[Union[Array[("num_samples",), np.float64], int]] = None,
+        transpose_and_squeeze=True,
     ) -> Tuple[
         Array[("num_samples",), np.float64],
-        Array[("num_samples", "num_y_channels"), np.float64],
+        Array[("num_y_channels", "num_samples"), np.float64],
     ]:
         """Evaluates the model function using the fitted parameter set.
 
         :param x_fit: optional x-axis points to evaluate the model at. If `None` we use
             the dataset values. If a scalar we generate an axis of linearly spaced
-            points between the minimum and maxim value of the x-axis dataset. Otherwise
-            it should be an array of x-axis data points to use.
+            points between the minimum and maximum value of the x-axis dataset.
+            Otherwise it should be an array of x-axis data points to use.
```

Comment:
Can you document `transpose_and_squeeze` please? We should explicitly mention plotting here. e.g. say one can do `plt.plot(*fit.eval(True))`

Should we put this before `x_fit`? It is a breaking change which is bad, bit it makes the most common use-case (plotting) easier since the user does not need to do `evaluate(None, True)`.

Could you update the start of the README which has a plotting example please? And also check for uses within `ionics_fits` of `evaluate` which may need an update (I think it's used in the tests).

Reply:
I'd be in favour of keeping the signature as it is currently, people can already do `plt.plot(*fit.eval(tranpose_and_squeeze=True))`. I would say the main parameter of this method is `x_fit` and would put this as the first parameter therefore.

Reply:
> I would say the main parameter of this method is x_fit and would put this as the first parameter therefore.

Well, I guess this is the main question. To me the most common use case is `plt.plot(*fit.eval(transpose_and_squeeze=True))` rather than providing `x_fit`, but if that's not the case I agree with you. And, even when one does provide `x_fit` one is probably still setting `transpose_and_squeeze` to `True`.

What I don't like about the current structure is that the most common use of this method requires me to remember a somewhat verbose argument name, which is a shame.

Reply:
> Since the default for transpose_and_squeeze is True

I hadn't appreciated this. I think this is a bad idea.

IMHO it's really important for libraries to behave consistently internally, this means that the default shapes used by functions should all be the same. Having options to reshape is useful, but I think it should never be default as it's just too much of a footgun.